src/nvim/ex_docmd.c

```diff
@@ -6788,27 +6801,51 @@ void free_cd_dir(void)
 
 #endif
 
-/*
- * Deal with the side effects of changing the current directory.
- * When "local" is TRUE then this was after an ":lcd" command.
- */
-void post_chdir(int local)
+/// Deal with the side effects of changing the current directory.
+///
+/// @param scope  Scope of the function call (editor, tab or window).
+void post_chdir(int scope)
 {
+  assert(scope >= kCdScopeEditor && scope <= MAX_CD_SCOPE);
+
+  enum cd_scope s = scope; // Treat it as an enum so switch catches unused cases.
```

Comment:
I'm hitting my head against a wall now (figuratively), you are right. It's building now and the new commits are pushed.